docs/content/manual/manual.md

```diff
@@ -0,0 +1,5538 @@
+### `JOIN($idx; stream; idx_expr; join_expr)`
+
+  This builtin joins the values from the given stream to the
+  given index.  The index's keys are computed by applying the
+  given index expression to each value from the given stream.
+  An array of the value in the stream and the corresponding
+  value from the index is fed to the given join expression to
+  produce each result.
+
+### `JOIN($idx; stream; idx_expr)`
+
+  Same as `JOIN($idx; stream; idx_expr; .)`.
+
+### `JOIN($idx; idx_expr)`
+
+  This builtin joins the input `.` to the given index, applying
+  the given index expression to `.` to compute the index key.
+  The join operation is as described above.
+
+### `IN(s)`
+
+  This builtin outputs `true` if `.` appears in the given
+  stream, otherwise it outputs `false`.
+
+### `IN(source; s)`
+
+  This builtin outputs `true` if any value in the source stream
+  appears in the second stream, otherwise it outputs `false`.
+
+## Regular expression functions
+
+jq uses the
+[Oniguruma regular expression library](https://github.com/kkos/oniguruma/blob/master/doc/RE),
+as do PHP, TextMate, Sublime Text, etc, so the
+description here will focus on jq specifics.
+
+Oniguruma supports several flavors of regular expression, so it is important to know
+that jq uses the ["Perl NG" (Perl with named groups)](https://github.com/kkos/oniguruma/blob/master/doc/SYNTAX.md) flavor.
+
+The jq regex filters are defined so that they can be used using
+one of these patterns:
+
+    STRING | FILTER(REGEX)
+    STRING | FILTER(REGEX; FLAGS)
+    STRING | FILTER([REGEX])
+    STRING | FILTER([REGEX, FLAGS])
+
+where:
+
+* STRING, REGEX, and FLAGS are jq strings and subject to jq string interpolation;
+* REGEX, after string interpolation, should be a valid regular expression;
+* FILTER is one of `test`, `match`, or `capture`, as described below.
+
+Since REGEX must evaluate to a JSON string, some characters that are needed
+to form a regular expression must be escaped. For example, the regular expression
+`\s` signifying a whitespace character would be written as `"\\s"`.
+
+FLAGS is a string consisting of one of more of the supported flags:
+
+* `g` - Global search (find all matches, not just the first)
+* `i` - Case insensitive search
+* `m` - Multi line mode (`.` will match newlines)
+* `n` - Ignore empty matches
+* `p` - Both s and m modes are enabled
+* `s` - Single line mode (`^` -> `\A`, `$` -> `\Z`)
+* `l` - Find longest possible matches
+* `x` - Extended regex format (ignore whitespace and comments)
+
+To match a whitespace with the `x` flag, use `\s`, e.g.
+
+    jq -n '"a b" | test("a\\sb"; "x")'
+
+Note that certain flags may also be specified within REGEX, e.g.
+
+    jq -n '("test", "TEst", "teST", "TEST") | test("(?i)te(?-i)st")'
+
+evaluates to `true`, `true`, `false`, `false`.
+
+::: Compatibility
+jaq uses the [`regex` library](https://docs.rs/regex/latest/regex/) instead of Oniguruma,
+which can result in subtle differences in regex execution.
```

Comment:
gojq uses the [`regexp` package](https://pkg.go.dev/regexp) so it is different from Oniguruma.

Reply:
Done.